### PR TITLE
Closes #6621 Increase phpstan level to level 1

### LINKED ISSUE #6621: Increase PHPStan to rule level to 1
#6541  introduced PhpStan and started with level 0 conformance. Now we can increase the rule to further improve the codebase.

Once this is resolved, another issue should be created to bump the rule level again.

### PR DESCRIPTION
**Pull request by Miraeld**
# Description

Fixes #6621 

### User documentation

It won't impact users. 

### Technical documentation

This PR simply changes some things on how the code is structured to comply with php-stan. 

## Type of change
- [x] Enhancement (non-breaking change which improves an existing functionality).

## Risks

# Checklists

## Feature validation

- [x] I validated all the Acceptance Criteria. If possible, provide sreenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Documentation

- [x] I prepared the user documentation for the feature/enhancement and shared it in the PR or the GitHub issue.
- [x] The user documentation covers new/changed entry points (endpoints, WP hooks, configuration files, ...).
- [x] I prepared the technical documentation if needed, and shared it in the PR or the GitHub issue.

## Code style
- [x] I wrote self-explanatory code about what it does.
- [x] I wrote comments to explain why it does it.
- [x] I named variables and functions explicitely.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unecessary complexity.
- [x] I listed the introduced external dependencies explicitely on the PR.
- [x] I validated the repo-specific guidelines from CONTRIBUTING.md.

## Observability
- [x] I handled errors when needed.
- [x] I wrote user-facing messages that are understandable and provide actionable feedbacks.
- [x] I prepared ways to observe the implemented system (logs, data, etc.).

## Risks
- [x] I explicitely mentioned performance risks in the PR.
- [x] I explicitely mentioned security risks in the PR.

### CODE REVIEW

File: inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php [hunk 1]
```diff
@@ -511,6 +511,7 @@ function ( $item ) {
 				);
 
 				// Check if the font URL matches any part of the exclude_fonts_preload array.
+				// @phpstan-ignore-next-line .
```
Comment (remyperona) [inline inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php]: `Variable $exclude_fonts_preload_pattern in empty() always exists and is not falsy.  `

We could handle that without ignoring no? Maybe using `array_filter()` before the `implode()`

File: inc/ThirdParty/Hostings/LiteSpeed.php [hunk 2]
```diff
@@ -39,7 +39,7 @@ public function litespeed_send_headers() {
 			return;
 		}
 		foreach ( $this->headers as $header ) {
-			@header( "X-LiteSpeed-Purge: {$header}", false ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
+			@header( "X-LiteSpeed-Purge: {$header}" ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
```
Comment (remyperona) [inline inc/ThirdParty/Hostings/LiteSpeed.php]: this should not be modified, it's a bug caused by a fixture overridding the default header function for tests. The fixture should be updated instead to accept the second optional parameter and match the original (override_header_functions.php)

File: inc/ThirdParty/Plugins/SEO/SEOPress.php [hunk 3]
```diff
@@ -27,7 +27,7 @@ public function __construct( Options_Data $option ) {
 	 * Subscribed events.
 	 */
 	public static function get_subscribed_events() {
-		if ( ! function_exists( 'seopress_get_toggle_option' ) || 1 !== (int) seopress_get_toggle_option( 'xml-sitemap' ) ) {
+		if ( ! function_exists( 'seopress_get_toggle_option' ) || 1 !== (int) seopress_get_toggle_option() ) {
```
Comment (remyperona) [inline inc/ThirdParty/Plugins/SEO/SEOPress.php]: same thing here, the report is caused by the fixture not correctly matching the original

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/ThirdParty/Hostings/LiteSpeed.php (hunk 2 above)
Comment (Miraeld) [reply inc/ThirdParty/Hostings/LiteSpeed.php]: Modified

File: inc/ThirdParty/Plugins/SEO/SEOPress.php (hunk 3 above)
Comment (Miraeld) [reply inc/ThirdParty/Plugins/SEO/SEOPress.php]: Modified.

File: inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php (hunk 1 above)
Comment (Miraeld) [reply inc/Engine/Optimization/RUCSS/Controller/UsedCSS.php]: Done

File: phpstan-bootstrap.php
Comment (remyperona) [inline phpstan-bootstrap.php:1]: We could move this in the `tests/phpstan` directory instead

Comment (Miraeld) [reply phpstan-bootstrap.php:1]: It's been moved :)

**Review by remyperona — APPROVED** [review]

**Comment by MathieuLamiot** [comment]
@Miraeld can this be merged and moved to Done?

**Pull request merged**